app/controllers/app.py

```diff
@@ -1781,6 +1781,121 @@ def post(self, app_id):
             return dict(status='fail', message=str(e)), 500
 
 
+class AppDockerListenerView(Resource):
+    def post(self):
+        """
+        Redeploy application from the Docker Hub webhook payload 
+        """
+        payload = request.get_json()
+        repository = payload['repository']['repo_name']
+
+        app_image = f"{repository}:{payload['push_data']['tag']}"
+
+        apps = App.find_all(image=repository)
+        if not apps:
+            return dict(
+                status="fail",
+                message=f"App with image {app_image} not found"
+            ), 404
+        try:
+            for app in apps:
+                if app.disabled or app.project.disabled:
+                    continue
+                project = app.project
+
+                if not project:
+                    log_activity('App', status='Failed',
+                                 operation='Update',
+                                 description='project not found',
+                                 a_project_id=project.id,
+                                 a_cluster_id=project.cluster_id,
+                                 a_app_id=app.id)
+                    continue
+
+                cluster = project.cluster
+                namespace = project.alias
+
+                if not cluster or not namespace:
+                    log_activity('App', status='Failed',
+                                 operation='Update',
+                                 description='Cluster or namespace not found',
+                                 a_project_id=project.id,
+                                 a_cluster_id=project.cluster_id,
+                                 a_app_id=app.id)
+                    continue
+
+                kube_host = cluster.host
+                kube_token = cluster.token
+
+                kube_client = create_kube_clients(kube_host, kube_token)
+
+                dep_name = f'{app.alias}-deployment'
+
+                cluster_deployment = kube_client.appsv1_api.read_namespaced_deployment(
+                    name=dep_name,
+                    namespace=namespace
+                )
+                if app.private_image:
+                    try:
+                        app_secret = kube_client.kube.read_namespaced_secret(
+                            namespace=namespace,
+                            name=app.alias
+                        )
+
+                    except Exception as e:
+                        print(e)
```

Comment:
Just clarification on this . Instead of printing the exception to the console directly , cant we add it to the logs for better tracking ? Whats your suggestion on this ?

Reply:
Great catch, I had eliminated the logs because we have no component to attach them to, but I think we can make them work